packages/sdk/src/light_push/light_push.ts

```diff
@@ -81,7 +83,11 @@ export class LightPush implements ILightPush {
       ...options
     };
 
-    const { pubsubTopic } = encoder;
+    const pubsubTopic = determinePubsubTopic(
+      encoder.contentTopic,
+      this.protocol.clusterId,
+      encoder.pubsubTopicOrShard
+    );
```

Comment:
@weboko we discussed that you'd prefer for this to remains in the encoder side, even if it clutters the `createEncoder` API as the intent would be to pass most of the context from the SDK's `Waku` to keep `Waku.createEncoder` simple.